### PR TITLE
log refactor: load and dump

### PR DESCRIPTION
**Pull request by GregoryIan**
### What problem does this PR solve? 
refactor log of loader and mydumper

### What is changed and how it works?
use pingcap/log

Pr recommended review order
https://github.com/pingcap/dm/pull/162
https://github.com/pingcap/dm/pull/170
https://github.com/pingcap/dm/pull/172
https://github.com/pingcap/dm/pull/173
https://github.com/pingcap/dm/pull/175
https://github.com/pingcap/dm/pull/176

### CODE REVIEW

File: dm/master/sql-operator/operator.go [hunk 1]
```diff
@@ -108,14 +111,14 @@ func (h *Holder) Set(req *pb.HandleSQLsRequest) error {
 	if ok1 {
 		prev, ok2 := operators[key]
 		if ok2 {
-			log.Warnf("[sql-operator] overwrite previous operator %s by operator %s", prev, oper)
+			h.logger.Warn(" overwrite operation", zap.Stringer("previous operation", prev), zap.Stringer("current operation", oper))
```
Comment (WangXiangUSTC) [inline dm/master/sql-operator/operator.go]:
```suggestion
			h.logger.Warn("overwrite operation", zap.Stringer("previous operation", prev), zap.Stringer("current operation", oper))
```

File: loader/db.go [hunk 2]
```diff
@@ -158,30 +159,30 @@ func executeSQLImp(db *sql.DB, sqls []string) error {
...
-		log.Debugf("[exec][sql]%-.200v", sqls[i])
+		logger.Debug("execute statement", zap.String("sqls", fmt.Sprintf("%-.200v", sqls[i])))
 		res, err = txn.Exec(sqls[i])
 		if err != nil {
-			log.Warnf("[exec][sql]%-.100v[error]%v", sqls[i], err)
+			logger.Warn("execute statement", zap.String("sqls", fmt.Sprintf("%-.200v", sqls[i])), log.ShortError(err))
 			rerr := txn.Rollback()
 			if rerr != nil {
-				log.Errorf("[exec][sql]%-.100s[error]%v", sqls, rerr)
+				logger.Error("fail rollback", zap.Error(rerr))
```
Comment (WangXiangUSTC) [inline loader/db.go]: how about use ShortError

File: loader/loader.go [hunk 3]
```diff
@@ -124,7 +129,7 @@ func (w *Worker) run(ctx context.Context, fileJobQueue chan *fileJob, workerWg *
 		for {
 			select {
 			case <-newCtx.Done():
-				log.Debugf("[loader] worker %d execution goroutine exits", w.id)
+				w.logger.Debug("execution goroutine exits")
```
Comment (WangXiangUSTC) [inline loader/loader.go]: forget add id info?

File: loader/loader.go [hunk 4]
```diff
@@ -842,7 +849,7 @@ func (l *Loader) restoreTable(conn *Conn, sqlFile, schema, table string) error {
 	err := l.restoreStructure(conn, sqlFile, schema, table)
 	if err != nil {
 		if isErrTableExists(err) {
-			log.Infof("[loader][table already exists, skip]%s", sqlFile)
+			l.logger.Info("able already exists, skip it", zap.String("sql", sqlFile))
```
Comment (WangXiangUSTC) [inline loader/loader.go]:
```suggestion
			l.logger.Info("table already exists, skip it", zap.String("sql", sqlFile))
```

File: loader/loader.go [hunk 5]
```diff
@@ -913,15 +920,15 @@ func renameShardingSchema(query, srcSchema, dstSchema string) string {
...
+func fetchMatchedLiteral(logger log.Logger, router *router.Table, schema, table string) (targetSchema string, targetTable string) {
 	if schema == "" {
 		// nothing change
 		return schema, table
 	}
 
 	targetSchema, targetTable, err := router.Route(schema, table)
 	if err != nil {
-		log.Error(errors.ErrorStack(err)) // log the error, but still continue
+		logger.Error("fail to route table", zap.String("unit", "loader")) // log the error, but still continue
```
Comment (WangXiangUSTC) [inline loader/loader.go]: forget error info

File: loader/convert_data.go [hunk 6]
```diff
@@ -227,7 +228,7 @@ func tableName(schema, table string) string {
 	return fmt.Sprintf("`%s`.`%s`", schema, table)
 }
 
-func parseTable(r *router.Table, schema, table, file string) (*tableInfo, error) {
+func parseTable(logger log.Logger, r *router.Table, schema, table, file string) (*tableInfo, error) {
```
Comment (WangXiangUSTC) [inline loader/convert_data.go]: put log in function's args seems ugly... have any other way?

File: loader/loader.go (hunk 3 above)
Comment (GregoryIan) [reply loader/loader.go]: logger contains it

File: loader/convert_data.go (hunk 6 above)
Comment (GregoryIan) [reply loader/convert_data.go]: think logger as session context, in fact I develop a context pkg and still don't use it now, any suggestions?

**Pull request closed without merging**